moveit_ros/planning/trajectory_execution_manager/test/test_execution_manager.cpp

```diff
@@ -1,107 +1,103 @@
- *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- *  POSSIBILITY OF SUCH DAMAGE.
- *********************************************************************/
-
-/* Author: Ioan Sucan */
-
-#include <moveit/trajectory_execution_manager/trajectory_execution_manager.h>
-#include <moveit/robot_model_loader/robot_model_loader.h>
-#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+/* Author: Cristian C. Beltran
+   Desc: Test the TrajectoryExecutionManager with MoveitCpp
+*/
 
-int main(int argc, char** argv)
-{
-  ros::init(argc, argv, "test_trajectory_execution_manager");
-
-  ros::AsyncSpinner spinner(1);
-  spinner.start();
-
-  robot_model_loader::RobotModelLoaderPtr rml(new robot_model_loader::RobotModelLoader);
-  planning_scene_monitor::PlanningSceneMonitor psm(rml);
-  trajectory_execution_manager::TrajectoryExecutionManager tem(rml->getModel(), psm.getStateMonitor(), true);
+// ROS
+#include <ros/ros.h>
 
-  std::cout << "1:\n";
-  if (!tem.ensureActiveControllersForJoints(std::vector<std::string>(1, "basej")))
-    ROS_ERROR("Fail!");
+// Testing
+#include <gtest/gtest.h>
 
-  std::cout << "2:\n";
-  if (!tem.ensureActiveController("arms"))
-    ROS_ERROR("Fail!");
+// Main class
+#include <moveit/moveit_cpp/moveit_cpp.h>
+#include <moveit/moveit_cpp/planning_component.h>
+// Msgs
+#include <geometry_msgs/PointStamped.h>
 
-  std::cout << "3:\n";
-  if (!tem.ensureActiveControllersForJoints(std::vector<std::string>(1, "rj2")))
-    ROS_ERROR("Fail!");
+namespace moveit_cpp
+{
+class MoveItCppTest : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    nh_ = ros::NodeHandle("/test_execution_manager");
+    moveit_cpp_ptr = std::make_shared<MoveItCpp>(nh_);
+    trajectory_execution_manager_ptr = moveit_cpp_ptr->getTrajectoryExecutionManager();
+
+    traj1.joint_trajectory.joint_names.push_back("panda_joint1");
+    traj1.joint_trajectory.points.resize(1);
+    traj1.joint_trajectory.points[0].positions.push_back(0.0);
+
+    traj2 = traj1;
+    traj2.joint_trajectory.joint_names.push_back("panda_joint2");
+    traj2.joint_trajectory.points[0].positions.push_back(1.0);
+    traj2.multi_dof_joint_trajectory.joint_names.push_back("panda_joint3");
+    traj2.multi_dof_joint_trajectory.points.resize(1);
+    traj2.multi_dof_joint_trajectory.points[0].transforms.resize(1);
+  }
+
+protected:
+  ros::NodeHandle nh_;
+  MoveItCppPtr moveit_cpp_ptr;
+  PlanningComponentPtr planning_component_ptr;
+  trajectory_execution_manager::TrajectoryExecutionManagerPtr trajectory_execution_manager_ptr;
+  moveit_msgs::RobotTrajectory traj1;
+  moveit_msgs::RobotTrajectory traj2;
+};
 
-  std::cout << "4:\n";
-  if (!tem.ensureActiveControllersForJoints(std::vector<std::string>(1, "lj1")))
-    ROS_ERROR("Fail!");
+TEST_F(MoveItCppTest, EnsureActiveControllersForJointsTest)
+{
+  ASSERT_TRUE(
+      trajectory_execution_manager_ptr->ensureActiveControllersForJoints(std::vector<std::string>(1, "panda_joint1")));
```

Comment:
Can't you just write:
```suggestion
      trajectory_execution_manager_ptr->ensureActiveControllersForJoints({"panda_joint1"});
```

Reply:
No, the ensureActiveCountrollersForJoints is only defined for a vector of strings. I could add an overload for just a string if needed

Reply:
The angle brackets create an initializer list which should be implicitly converted to a vector

Reply:
Simon is right, but he refers to the curly brackets actually :wink:
Please try the suggestion.

Reply:
Oh sorry, I didn't notice the curly brackets the first time. I thought you were referring to the string alone.